### ISSUE TITLE
Enhance configuration documentation for version/channel

### ISSUE THREAD

**Issue by calebcartwright** [rust-lang/rustfmt#4178]
Refs https://github.com/rust-lang/rustfmt/issues/4128#issuecomment-622985809 and https://github.com/rust-lang/rustfmt/issues/4128#issuecomment-629609234

The configuration options for rustfmt are defined within this repo in [Configurations.md](https://github.com/rust-lang/rustfmt/blob/master/Configurations.md) and are also available on [GH pages](http://rust-lang.github.io/rustfmt/)

However, these configuration option listings are reflective of the latest configuration settings within source control on the master branch. As such this can drift from the options available in released versions of rustfmt which has understandably been a source of confusion for users.

We should enhance the public documentation of rustfmt configuration options to take version and channel/distribution into account so that users will be able to determine the true set of config options available for the version of rustfmt they are using.

**Comment by ayazhafiz** [comment]
One way to do this would be to backfill an added-on-version attribute to each config option, similar to [rustc's `since`](https://github.com/rust-lang/rust/blob/e5335592e78354e33d798d20c04bcd677c1df62d/src/liballoc/vec.rs#L60). Then new config options always take on the version of the next planned release (the problem here is a release must be committed to or some cleanup work should be done before a release). With this information we could parse the configuration readme into several files, perhaps one per minor version and channel (stable/beta for stable features, nightly for all), which get put on GH pages. Readers of the config readme still get told whether a feature is stable and what version it was introduced in, which should be enough information for determining whether the feature is available.

@calebcartwright what do you think?

It would also be nice if we could get rustdoc/docs.rs to do a lot of this work, but I don't think that's possible without exposing other library information.

**Comment by ayazhafiz** [comment]
I like this idea. Looks like the github page takes a tagged version of the config file to render

https://github.com/rust-lang/rustfmt/blob/e2547530720a136c47e4bd0d1be1bb5e7ba09c34/docs/index.html#L69

so adding a dropdown (or similar) to change the version should be straightforward. Also looks like gh page has a toggle for stable/all features, but it doesn't seem to work correctly (at least on my firefox)

**Comment by calebcartwright** [comment]
I really like the solution we've implemented, and it feels like it's covered the major use case that motivated this issue. I've got a couple other smaller feature requests/nice-to-haves, but will open up separate issues for those.
